snowflake/pkg/utils/file/file.go

```diff
@@ -78,7 +80,7 @@ func DownloadAndUntar(ctx context.Context, logger logr.Logger, url string, dir s
 		if err != nil {
 			return err
 		}
-		dest := filepath.Join(dir, hdr.Name)
+		dest := filepath.Join(dir, filepath.Clean(hdr.Name))
```

Comment:
I don't think clean will help as per the example here https://www.geeksforgeeks.org/filepath-clean-function-in-golang-with-examples/,
we should eliminate any ".." with current path. Please correct me if I am wrong

Reply:
Thanks for pointing out. Yes Clean() can't eliminate all ".." in the path, but I believe we are not expected to change the path in this case.
Therefore, I added an error return when ".." is still detected after Clean().

Reply:
I guess the new check will be sufficient.